src/index.js

```diff
@@ -343,32 +356,37 @@ class IPLDResolver {
 
       // End of iteration if there aren't any paths left to return or
       // if we don't want to traverse recursively and have already
-      // returne the first level
+      // returns the first level
       while (treePaths.length > 0 || queue.length > 0) {
         // There aren't any paths left, get them from the given CID
         if (treePaths.length === 0 && queue.length > 0) {
-          ({ cid, basePath } = queue.shift())
-          const format = await this.getFormat(cid.codec)
-          block = await this.bs.get(cid, options)
+          const next = queue.shift()
+
+          if (next) {
```

Comment:
I guess this is needed due to type checking restrictions?

Reply:
`queue.shift()` can return `undefined` if `queue` is empty so the complier complains if we don't handle that case.